algorithms/src/sorting/Kokkos_SortPublicAPI.hpp

```diff
@@ -53,13 +53,9 @@ void sort(const ExecutionSpace& exec,
 
   if constexpr (Impl::better_off_calling_std_sort_v<ExecutionSpace>) {
     exec.fence("Kokkos::sort without comparator use std::sort");
-    if (view.span_is_contiguous()) {
-      std::sort(view.data(), view.data() + view.size());
-    } else {
-      auto first = ::Kokkos::Experimental::begin(view);
-      auto last  = ::Kokkos::Experimental::end(view);
-      std::sort(first, last);
-    }
+    auto first = ::Kokkos::Experimental::begin(view);
+    auto last  = ::Kokkos::Experimental::end(view);
+    std::sort(first, last);
```

Comment:
Not sure this is a good idea. I wouldn't be at ALL surprised if std::sort with pointers is faster/specialized than with arbitrary iterators.